File: 01-basic-challenges-1/08-palindrome/palindrome.js
function isPalindrome(string) {
  // only lowercase letters or digits (no spaces, no punctuation)
  const cleanString = string.toLowerCase().replace(/[^a-z0-9]/g, "");
  const reversedClean = cleanString.split("").reverse().join("");
  return reversedClean === cleanString;
}

module.exports = isPalindrome;

File: 01-basic-challenges-1/08-palindrome/palindrome.rb
def is_palindrome?(string)
  # If it's not lowercase character or number, remove
  clean_string = string.downcase.gsub(/[^a-z0-9]/, "")
  clean_string == clean_string.reverse
end

puts is_palindrome?("racecar")
puts is_palindrome?("racecars")
